homeassistant/helpers/trigger.py

```diff
@@ -621,3 +625,103 @@ async def async_get_all_descriptions(
 
     hass.data[TRIGGER_DESCRIPTION_CACHE] = new_descriptions_cache
     return new_descriptions_cache
+
+
+class TargetSelectorStateChangeTracker:
```

Comment:
I'd suggest to name the class `TargetStateChangeTracker` with the rationale that a target _selector_ is a way to ask users to input a target, this class allows listening to state change events matching a target, regardless of if that target was defined by a target selector or in some other way.